tests/integration/_experimental/dataset/test_dataset.py

```diff
@@ -169,3 +177,92 @@ def test__fetch_dataset__not_exist() -> None:
     name = with_test_prefix(f"{__file__}::test__fetch_dataset__not_exist")
     with pytest.raises(NotFoundError):
         fetch_dataset(name)
+
+
+@pytest.fixture(scope="module")
+def with_dataset_commits() -> Tuple[int, List[CommitData]]:
+    for version in range(TEST_COMMIT_HISTORY_VERSIONS):
+        # remove all previous datapoints and add (version + 1) new datapoints in each iteration
+        datapoints = [dict(locator=f"{version}-{i}") for i in range(version + 1)]
+        register_dataset(TEST_COMMIT_HISTORY_NAME, pd.DataFrame(datapoints), id_fields=["locator"])
+    return fetch_commits(TEST_COMMIT_HISTORY_NAME)
+
+
+def test__fetch_commits(with_dataset_commits: Tuple[int, List[CommitData]]) -> None:
+    # check fetch_commits without optional args
+    total_commits, commits = with_dataset_commits
+    assert total_commits == TEST_COMMIT_HISTORY_VERSIONS
+    previous_commit_time = -1
+    for version, commit in enumerate(commits):
+        assert commit.timestamp >= previous_commit_time
+        assert commit.n_removed == version
+        assert commit.n_added == version + 1
+        previous_commit_time = commit.timestamp
+
+
+@pytest.mark.depends(on=["test__fetch_commits"])
```

Comment:
Do we this depends? Looks like `test__fetch_commits` is not doing modification to data `with_dataset_commits` and these tests can just run without the dependency.

Reply:
Right we don't need this dependency marked explicitly. My mental model was to skip other tests if `test__fetch_commits` already fails, but there is no logical dependency between them